test/e2e/runner/load.go

```diff
@@ -172,10 +167,21 @@ func createTxBatch(ctx context.Context, txCh chan<- types.Tx, testnet *e2e.Testn
 					panic(fmt.Sprintf("Failed to generate tx: %v", err))
 				}
 
-				select {
-				case txCh <- tx:
-				case <-ctx.Done():
-					return
+				var nodeIndices []int
+				if testnet.LoadNumNodesPerTx <= 1 {
+					// Pick one random node to send the transaction.
+					nodeIndices = []int{cmtrand.Intn(len(targetNodes))}
+				} else {
+					// Pick LoadNumNodesPerTx random nodes (channel indices) to
+					// send the transaction.
+					nodeIndices = cmtrand.Perm(len(targetNodes))[:testnet.LoadNumNodesPerTx]
```

Comment:
TODO: If LoadNumNodesPerTx is too small compared to len(targetNodes), it may be worth to pick LoadNumNodesPerTx random elements one by one from targetNodes instead of doing a permutation.